tests/tokenWrapper.test.ts

```diff
@@ -0,0 +1,800 @@
+        console.log(`new proposalNonce is ${isValidAddressAgain.output}`);
+
+        // validate that proposalNonce has increased
+        let  newProposalNonceAgain  = await tokenWrapperContract.query.nonce();
+        console.log(`new proposalNonce is ${newProposalNonceAgain.output}`);
+        expect(newProposalNonceAgain.output).to.be.equal(proposalNonce);
+    });
+
+    it('Update config', async () => {
+        const {contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit,} = tokenWrapperContractInitParams(sender, BobSigner, CharlieSigner)
+
+        let newGovernor = BobSigner.address;
+        let newIsNativeAllowed = false;
+        let newWrappingLimit = 1000;
+        let newFeePercentage = 2;
+        let newFeeRecipient = FerdieSigner.address;
+
+        // update config with new states
+        let updateConfigFunction = await tokenWrapperContract.tx.updateConfig(newGovernor, newIsNativeAllowed, newWrappingLimit,
+            newFeePercentage, newFeeRecipient)
+        expect(updateConfigFunction).to.be.ok;
+
+        // validate that new governor is not same as old governor
+        let newGovernorFromStorage = await tokenWrapperContract.query.governor();
+        expect(newGovernorFromStorage.output).to.equal(newGovernor);
+        expect(newGovernorFromStorage.output).to.not.equal(contractGovernor);
+
+        // validate that isNativeAllowed has changed
+        let  newIsNativeAllowedFromStorage = await tokenWrapperContract.query.isNativeAllowed();
+        expect(newIsNativeAllowedFromStorage.output).to.be.equal(newIsNativeAllowed);
+        expect(newIsNativeAllowedFromStorage.output).to.not.equal(isNativeAllowed);
+
+        // validate that isWrappingLimit has changed
+        let newWrappingLimitFromStorage = await tokenWrapperContract.query.wrappingLimit();
+        expect(Number(newWrappingLimitFromStorage.output)).to.equal(newWrappingLimit);
+        expect(Number(newWrappingLimitFromStorage.output)).to.not.equal(wrappingLimit);
+
+        // validate that feePercentage has changed
+        let newFeePercentageFromStorage = await tokenWrapperContract.query.feePercentage();
+        expect(Number(newFeePercentageFromStorage.output)).to.equal(newFeePercentage);
+        expect(Number(newFeePercentageFromStorage.output)).to.not.equal(feePercentage);
+
+        // validate that feeRecipient has changed
+        let newFeeRecipientFromStorage = await tokenWrapperContract.query.feeRecipient();
+        expect(newFeeRecipientFromStorage.output).to.equal(newFeeRecipient);
+        expect(newFeeRecipientFromStorage.output).to.not.equal(feeRecipient);
+
+    });
+
+    it('Test native wrapping functionality', async () => {
+        let initialSenderWrappedBalance = await tokenWrapperContract.query.psp22Balance(sender.address);
+        let initialContractBalance = await tokenWrapperContract.query.nativeContractBalance();
+
+        expect(Number(initialSenderWrappedBalance.output)).to.not.equal(0);
```

Comment:
I'm confused why the initial sender wrapped balance is not 0? What should it be, where did the first wrapping occur?